src/OpenTelemetry/Instrumentation/PropertyFetcher.cs

```diff
@@ -13,13 +13,14 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 // </copyright>
+
 using System;
 using System.Linq;
 using System.Reflection;
 
-namespace OpenTelemetry.Instrumentation.Http.Implementation
+namespace OpenTelemetry.Instrumentation
 {
-    internal class PropertyFetcher
+    public class PropertyFetcher
```

Comment:
I think we could improve this API. Internally it has a very nice, typed delegate created. But we only expose `object Fetch(object)` which basically throws away that typing information and will box any value types + require a cast. With a little work we could probably also expose:

```csharp
class PropertyFetcher<TType, TPropertyType> : PropertyFetcher
{
   public TPropertyType Fetch(TType instance)
}
```
That would be useful when you know the types ahead of time, which is actually the more common case. It would allow us to avoid some of those performance penalties.

Let's merge this and then we can try to improve it?

Reply:
sounds great :)

Reply:
@CodeBlanch not sure how it was done originally, but inside the same class, we already have a typedPropertyFetcher. but, in the end, we are just returning as an object.